lib/logger/lib/logger/translator.ex

```diff
@@ -79,22 +79,34 @@ defmodule Logger.Translator do
     {:ok, ["Application ", Atom.to_string(app), " exited: " | Application.format_error(reason)]}
   end
 
-  def translate(_min_level, :error, :format, message) do
+  def translate(
+        _min_level,
+        :error,
+        :report,
+        {{Task.Supervisor, :terminating},
+         %{
+           name: name,
+           starter: starter,
+           function: function,
+           args: args,
+           reason: reason
+         }}
+      ) do
     opts = Application.get_env(:logger, :translator_inspect_opts)
 
-    case message do
-      # TODO: Remove this once tasks are migrated to Logger
-      {'** Task ' ++ _, [name, starter, function, args, reason]} ->
-        {formatted, reason} = format_reason(reason)
-        metadata = [crash_reason: reason] ++ registered_name(name)
+    {formatted, reason} = format_reason(reason)
+    metadata = [crash_reason: reason] ++ registered_name(name)
 
-        msg =
-          ["Task #{inspect(name)} started from #{inspect(starter)} terminating"] ++
-            [formatted, "\nFunction: #{inspect(function, opts)}"] ++
-            ["\n    Args: #{inspect(args, opts)}"]
+    msg =
+      ["Task #{inspect(name)} started from #{inspect(starter)} terminating"] ++
+        [formatted, "\nFunction: #{inspect(function, opts)}"] ++
+        ["\n    Args: #{inspect(args, opts)}"]
 
-        {:ok, msg, metadata}
+    {:ok, msg, metadata}
+  end
```

Comment:
I have ported it as is, but is there any reason why do that in translator instead of using `report_cb` directly? I think it would make more sense to have uniform error reports.

Reply:
that Is a separate discussion. Unifying reports would be a good option but we would definitely have to push some things uostream before. For example, we include more information on debug level.

Reply:
@josevalim you mean in `inspect`? This is what 2nd argument in `report_cb` is for, to pass additional informations to formatters.

Reply:
I mean we show more information in general. Compare both reports in debug mode for a crashed GenServer and you will see the differences. :)